gctscript/vm/manager.go

```diff
@@ -0,0 +1,124 @@
+package vm
+
+import (
+	"errors"
+	"fmt"
+	"path/filepath"
+	"sync"
+	"sync/atomic"
+
+	"github.com/thrasher-corp/gocryptotrader/common"
+	"github.com/thrasher-corp/gocryptotrader/engine/subsystem"
+	"github.com/thrasher-corp/gocryptotrader/log"
+)
+
+const gctscriptManagerName = "GCTScript"
+
+// GctScriptManager loads and runs GCT Tengo scripts
+type GctScriptManager struct {
+	config   *Config
+	started  int32
+	stopped  int32
+	shutdown chan struct{}
+	// Optional values to override stored config ('nil' if not overridden)
+	Enabled            *bool
+	MaxVirtualMachines *uint8
+}
+
+// Started returns if gctscript manager subsystem is started
+func (g *GctScriptManager) Started() bool {
+	return atomic.LoadInt32(&g.started) == 1
+}
+
+// Start starts gctscript subsystem and creates shutdown channel
+func (g *GctScriptManager) Start(config *Config, wg *sync.WaitGroup) (err error) {
+	g.config = config
+	if !g.IsEnabled() {
+		return errors.New("manager has been disabled")
+	}
```

Comment:
Overwriting the config before performing any checks seems dangerous. It could already be running, this gets called and now the existing script manager has had all its settings changed and likely to have created a data race or a panic because nil was sent in.

Reply:
created a constructor/factory method for script manager